fragParsingError after backward seek in live sliding window stream

### What version of Hls.js are you using?

1.6.11, 1.6.12

### What browser (including version) are you using?

140.0.7339.133 (Official Build) (arm64)

### What OS (including version) are you using?

macOS 15.6

### Test stream

_No response_

### Configuration

```JavaScript
{
  "debug": true,
  "enableWorker": true,
  "lowLatencyMode": true,
  "backBufferLength": 90
}
```

### Additional player setup steps

No special setup steps required, this is reproducible on the default https://hlsjs.video-dev.org/demo/ page.

### Checklist

- [x] The issue observed is not already reported by searching on Github under https://github.com/video-dev/hls.js/issues
- [x] The issue occurs in the stable client (latest release) on https://hlsjs.video-dev.org/demo and not just on my page
- [x] The issue occurs in the latest client (main branch) on https://hlsjs-dev.video-dev.org/demo and not just on my page
- [x] The stream has correct Access-Control-Allow-Origin headers (CORS)
- [x] There are no network errors such as 404s in the browser console when trying to play the stream

### Steps to reproduce

1. Start playback of the live linear stream 
2. Seek back towards the start of the sliding window using the scrubber bar in the demo player
This is an intermittent issue that we last encounter  between 2025-09-11T09:00:00Z and 2025-09-11T13:00:00Z utc


### Expected behaviour

Player should seek to the position in the stream and resume playback

### What actually happened?

Player stalls with the following Error :

`Error: Found no media in fragment 11537 of level 0 resetting transmuxer to fallback to playlist timing at StreamController.updateLevelTiming`

### Console output

```shell
main.js:350 Using Hls.js config: {debug: true, enableWorker: true, lowLatencyMode: true, backBufferLength: 90}
logger.ts:102 [log] > Debug logs enabled for "Hls instance" in hls.js version 1.6.12
hls.ts:579 [log] > stopLoad
hls.ts:513 [log] > loadSource: -
interstitials-controller.ts:288 [log] > [interstitials]: clear schedule state
stream-controller.ts:601 [log] > [stream-controller]: Trigger BUFFER_RESET
hls.ts:466 [log] > attachMedia
buffer-controller.ts:320 [log] > [buffer-controller]: created media source: MediaSource
buffer-controller.ts:1546 [log] > [buffer-controller]: Media source opened
buffer-controller.ts:1375 [log] > [buffer-controller]: checkPendingTracks (pending: 0 codec events expected: 0) {}
level-controller.ts:351 [log] > [level-controller]: manifest loaded, 1 level(s) found, first bitrate: 0
buffer-controller.ts:266 [log] > [buffer-controller]: 1 bufferCodec event(s) expected.
playlist-loader.ts:397 [log] > auto startLoad with configured startPosition -1
hls.ts:556 [log] > startLoad(-1)
level-controller.ts:450 [log] > [level-controller]: Switching to level 0 (SDR @0) from level -1
base-playlist-controller.ts:346 [log] > [level-controller]: reload live playlist 0bps in 9725 ms
base-stream-controller.ts:2230 [log] > [stream-controller]: STOPPED->IDLE
base-stream-controller.ts:2230 [log] > [subtitle-stream-controller]: STOPPED->IDLE
base-playlist-controller.ts:216 [log] > [level-controller]: live playlist 0 REFRESHED 13328--1
base-playlist-controller.ts:346 [log] > [level-controller]: reload live playlist 0bps in 19724 ms
stream-controller.ts:661 [log] > [stream-controller]: Level 0 loaded [11477,13328][part-13328--1], cc [0, 0] duration:18007.46666667624
base-stream-controller.ts:1763 [log] > [stream-controller]: Live playlist sliding: 0.00 start-sn: 11477->11477 fragments: 1852
base-stream-controller.ts:1818 [log] > [stream-controller]: Setting startPosition to -1 to start at live edge 17977.478266773895
interstitials-controller.ts:1693 [log] > [interstitials]: INTERSTITIALS_UPDATED (19): ["1757586580_1095971179" 0.00-210.69],["1757587483_828786525" 735.66-915.63],["1757588208_3275627352" 1461.21-1641.13],["1757589360_1112304279" 2612.95-2735.27],["1757590199_3323689064" 3451.48-3661.48],["1757591128_2352494204" 4380.59-4560.55],["1757591808_3217355483" 5061.03-5241.00],["1757592868_650490437" 6120.70-6241.02],["1757593900_2452591813" 7152.51-7362.22],["1757594765_219942117" 8017.62-8197.59],["1757595408_4186765737" 8660.91-8840.83],["1757596540_3609787140" 9792.77-9912.70],["1757597300_3243867178" 10553.13-10763.09],["1757598283_2210540883" 11535.83-11715.67],["1757599008_1397443122" 12261.16-12441.17],["1757600262_4026601524" 13515.14-13727.32],["1757601780_209283036" 15033.00-15334.74],["1757602548_735391794" 15801.00-16040.96],["1757603906_4026256504" 17158.87-17369.60]
Schedule: ["1757586580_1095971179": 0.00-210.69],[primary: 210.69-735.66],["1757587483_828786525": 735.66-915.63],[primary: 915.63-1461.21],["1757588208_3275627352": 1461.21-1641.13],[primary: 1641.13-2612.95],["1757589360_1112304279": 2612.95-2735.27],[primary: 2735.27-3451.48],["1757590199_3323689064": 3451.48-3661.48],[primary: 3661.48-4380.59],["1757591128_2352494204": 4380.59-4560.55],[primary: 4560.55-5061.03],["1757591808_3217355483": 5061.03-5241.00],[primary: 5241.00-6120.70],["1757592868_650490437": 6120.70-6241.02],[primary: 6241.02-7152.51],["1757593900_2452591813": 7152.51-7362.22],[primary: 7362.22-8017.62],["1757594765_219942117": 8017.62-8197.59],[primary: 8197.59-8660.91],["1757595408_4186765737": 8660.91-8840.83],[primary: 8840.83-9792.77],["1757596540_3609787140": 9792.77-9912.70],[primary: 9912.70-10553.13],["1757597300_3243867178": 10553.13-10763.09],[primary: 10763.09-11535.83],["1757598283_2210540883": 11535.83-11715.67],[primary: 11715.67-12261.16],["1757599008_1397443122": 12261.16-12441.17],[primary: 12441.17-13515.14],["1757600262_4026601524": 13515.14-13727.32],[primary: 13727.32-15033.00],["1757601780_209283036": 15033.00-15334.74],[primary: 15334.74-15801.00],["1757602548_735391794": 15801.00-16040.96],[primary: 16040.96-17158.87],["1757603906_4026256504": 17158.87-17369.60],[primary: 17369.60-Infinity] pos: -1
interstitials-controller.ts:1036 [log] > [interstitials]: [checkStart] Advancing timeline position to 17977.478266773895
interstitials-controller.ts:1135 [log] > [interstitials]: setSchedulePosition 37, undefined ([primary: 17369.60-Infinity]) pos: 17977.478266773895
interstitials-controller.ts:2005 [log] > [interstitials]: INTERSTITIALS_BUFFERED_TO_BOUNDARY [primary: 17369.60-Infinity]
interstitials-controller.ts:1397 [log] > [interstitials]: resuming [primary: 17369.60-Infinity]
interstitials-controller.ts:1487 [log] > [interstitials]: [attachPrimary] Advancing timeline position to 17977.478266773895
buffer-controller.ts:1352 [log] > [buffer-controller]: Updating MediaSource duration to 18007.467
base-stream-controller.ts:1377 [log] > [stream-controller]: Setting startPosition to 17977.478266773895 to match start frag at live edge. mainStart: 17977.478266773895 liveSyncPosition: 17977.482266773895 frag.start: 17968.213333343236
base-stream-controller.ts:965 [log] > [stream-controller]: Loading main sn: 13325 of level 0 (frag:[17968.213-17978.027]) cc: 0 [11477-13328], target: 17977.478
base-stream-controller.ts:2230 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:88 [log] > injecting Web Worker for "main"
transmuxer-interface.ts:238 [log] > [transmuxer-interface]: Starting new transmux session for main sn: 13325 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: false
        timeOffset: 17968.213333343236
        initSegmentChange: true
base-stream-controller.ts:537 [log] > [stream-controller]: Loaded main sn: 13325 of level 0
bcc49293-4b96-4d2b-84ff-ad12f89764bd:1168 [log] > Debug logs enabled for "main" in hls.js version 1.6.12
transmuxer-interface.ts:396 [log] > ADTS sync word found !
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: reset next timestamp
bcc49293-4b96-4d2b-84ff-ad12f89764bd:11091 [log] > manifest codec:undefined, parsed codec:mp4a.40.2, channels:2, rate:48000 (ADTS object type:2 sampling index:3)
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: Found initPTS at playlist time: 17968.213333343236 offset: -9313.442333333332 (-838209810/90000) trackId: 2
base-stream-controller.ts:2230 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1490 [log] > [stream-controller]: Init audio buffer, container:audio/mp4, codecs[selected/level/parsed]=[mp4a.40.2//mp4a.40.2]
buffer-controller.ts:594 [log] > [buffer-controller]: BUFFER_CODECS: "audio" (current SB count 0)
buffer-controller.ts:1375 [log] > [buffer-controller]: checkPendingTracks (pending: 1 codec events expected: 1) {"audio":{"listeners":[],"codec":"mp4a.40.2","container":"audio/mp4","levelCodec":"mp4a.40.2","metadata":{"channelCount":2},"id":"main"}}
buffer-controller.ts:1444 [log] > [buffer-controller]: creating sourceBuffer(audio/mp4;codecs=mp4a.40.2) {"listeners":[],"codec":"mp4a.40.2","container":"audio/mp4","levelCodec":"mp4a.40.2","metadata":{"channelCount":2},"id":"main"}
buffer-controller.ts:1413 [log] > [buffer-controller]: SourceBuffers created. Running queue: 
video: (none) 
audio: (SourceBuffer) 
audiovideo: (none) }
buffer-controller.ts:952 [log] > [buffer-controller]: queuing "audio" append sn: 13325 of level 0 cc: 0
audio-stream-controller.ts:153 [log] > [audio-stream-controller]: InitPTS for cc: 0 found from main: -9313.442333333332 (-838209810/90000) trackId: 2
buffer-controller.ts:952 [log] > [buffer-controller]: queuing "audio" append sn: 13325 of level 0 cc: 0
transmuxer-interface.ts:396 [log] > [transmuxer.ts]: Flushed main sn: 13325 of level 0
base-stream-controller.ts:2230 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:2177 [log] > [stream-controller]: Parsed main sn: 13325 of level 0 (frag:[17968.213-17978.005])
buffer-controller.ts:1639 [log] > [buffer-controller]: Updating audio SourceBuffer timestampOffset to 9313.442333333332 (sn: 13325 cc: 0)
base-stream-controller.ts:768 [log] > [stream-controller]: Buffered main sn: 13325 of level 0 (frag:[17968.213-17978.005] > buffer:[17968.213-17978.027])
base-stream-controller.ts:2230 [log] > [stream-controller]: PARSED->IDLE
stream-controller.ts:1164 [log] > [stream-controller]: seek to target start position 17977.478266773895 from current time 0 buffer start 17968.213333
base-stream-controller.ts:965 [log] > [stream-controller]: Loading main sn: 13326 of level 0 (frag:[17978.005-17987.819]) cc: 0 [11477-13328], target: 17978.027
base-stream-controller.ts:2230 [log] > [stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:355 [log] > [stream-controller]: Media seeking to 17977.478, state: FRAG_LOADING, in buffer
base-stream-controller.ts:355 [log] > [audio-stream-controller]: Media seeking to 17977.478, state: STOPPED, in buffer
base-stream-controller.ts:355 [log] > [subtitle-stream-controller]: Media seeking to 17977.478, state: IDLE, in buffer
stream-controller.ts:573 [log] > [stream-controller]: Media seeked to 17977.478
base-stream-controller.ts:537 [log] > [stream-controller]: Loaded main sn: 13326 of level 0
base-stream-controller.ts:2230 [log] > [stream-controller]: FRAG_LOADING->PARSING
buffer-controller.ts:952 [log] > [buffer-controller]: queuing "audio" append sn: 13326 of level 0 cc: 0
transmuxer-interface.ts:396 [log] > [transmuxer.ts]: Flushed main sn: 13326 of level 0
base-stream-controller.ts:2230 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:2177 [log] > [stream-controller]: Parsed main sn: 13326 of level 0 (frag:[17978.027-17987.819])
base-stream-controller.ts:768 [log] > [stream-controller]: Buffered main sn: 13326 of level 0 (frag:[17978.027-17987.819] > buffer:[17968.213-17987.840])
base-stream-controller.ts:2230 [log] > [stream-controller]: PARSED->IDLE
base-stream-controller.ts:965 [log] > [stream-controller]: Loading main sn: 13327 of level 0 (frag:[17987.819-17997.632]) cc: 0 [11477-13328], target: 17987.84
base-stream-controller.ts:2230 [log] > [stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:537 [log] > [stream-controller]: Loaded main sn: 13327 of level 0
base-stream-controller.ts:2230 [log] > [stream-controller]: FRAG_LOADING->PARSING
buffer-controller.ts:952 [log] > [buffer-controller]: queuing "audio" append sn: 13327 of level 0 cc: 0
transmuxer-interface.ts:396 [log] > [transmuxer.ts]: Flushed main sn: 13327 of level 0
base-stream-controller.ts:2230 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:2177 [log] > [stream-controller]: Parsed main sn: 13327 of level 0 (frag:[17987.840-17997.632])
base-stream-controller.ts:768 [log] > [stream-controller]: Buffered main sn: 13327 of level 0 (frag:[17987.840-17997.632] > buffer:[17968.213-17997.653])
base-stream-controller.ts:2230 [log] > [stream-controller]: PARSED->IDLE
base-stream-controller.ts:965 [log] > [stream-controller]: Loading main sn: 13328 of level 0 (frag:[17997.632-18007.445]) cc: 0 [11477-13328], target: 17997.653
base-stream-controller.ts:2230 [log] > [stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:537 [log] > [stream-controller]: Loaded main sn: 13328 of level 0
base-stream-controller.ts:2230 [log] > [stream-controller]: FRAG_LOADING->PARSING
buffer-controller.ts:952 [log] > [buffer-controller]: queuing "audio" append sn: 13328 of level 0 cc: 0
transmuxer-interface.ts:396 [log] > [transmuxer.ts]: Flushed main sn: 13328 of level 0
base-stream-controller.ts:2230 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:2177 [log] > [stream-controller]: Parsed main sn: 13328 of level 0 (frag:[17997.653-18007.445])
base-stream-controller.ts:768 [log] > [stream-controller]: Buffered main sn: 13328 of level 0 (frag:[17997.653-18007.445] > buffer:[17968.213-18007.467])
base-stream-controller.ts:2230 [log] > [stream-controller]: PARSED->IDLE
favicon.ico:1  GET https://hlsjs.video-dev.org/favicon.ico 404 (Not Found)
interstitials-controller.ts:1693 [log] > [interstitials]: INTERSTITIALS_UPDATED (19): ["1757586580_1095971179" 0.00-210.69],["1757587483_828786525" 735.66-915.63],["1757588208_3275627352" 1461.21-1641.13],["1757589360_1112304279" 2612.95-2735.27],["1757590199_3323689064" 3451.48-3661.48],["1757591128_2352494204" 4380.59-4560.55],["1757591808_3217355483" 5061.03-5241.00],["1757592868_650490437" 6120.70-6241.02],["1757593900_2452591813" 7152.51-7362.22],["1757594765_219942117" 8017.62-8197.59],["1757595408_4186765737" 8660.91-8840.83],["1757596540_3609787140" 9792.77-9912.70],["1757597300_3243867178" 10553.13-10763.09],["1757598283_2210540883" 11535.83-11715.67],["1757599008_1397443122" 12261.16-12441.17],["1757600262_4026601524" 13515.14-13727.32],["1757601780_209283036" 15033.00-15334.74],["1757602548_735391794" 15801.00-16040.96],["1757603906_4026256504" 17158.87-17369.60]
Schedule: ["1757586580_1095971179": 0.00-210.69],[primary: 210.69-735.66],["1757587483_828786525": 735.66-915.63],[primary: 915.63-1461.21],["1757588208_3275627352": 1461.21-1641.13],[primary: 1641.13-2612.95],["1757589360_1112304279": 2612.95-2735.27],[primary: 2735.27-3451.48],["1757590199_3323689064": 3451.48-3661.48],[primary: 3661.48-4380.59],["1757591128_2352494204": 4380.59-4560.55],[primary: 4560.55-5061.03],["1757591808_3217355483": 5061.03-5241.00],[primary: 5241.00-6120.70],["1757592868_650490437": 6120.70-6241.02],[primary: 6241.02-7152.51],["1757593900_2452591813": 7152.51-7362.22],[primary: 7362.22-8017.62],["1757594765_219942117": 8017.62-8197.59],[primary: 8197.59-8660.91],["1757595408_4186765737": 8660.91-8840.83],[primary: 8840.83-9792.77],["1757596540_3609787140": 9792.77-9912.70],[primary: 9912.70-10553.13],["1757597300_3243867178": 10553.13-10763.09],[primary: 10763.09-11535.83],["1757598283_2210540883": 11535.83-11715.67],[primary: 11715.67-12261.16],["1757599008_1397443122": 12261.16-12441.17],[primary: 12441.17-13515.14],["1757600262_4026601524": 13515.14-13727.32],[primary: 13727.32-15033.00],["1757601780_209283036": 15033.00-15334.74],[primary: 15334.74-15801.00],["1757602548_735391794": 15801.00-16040.96],[primary: 16040.96-17158.87],["1757603906_4026256504": 17158.87-17369.60],[primary: 17369.60-Infinity] pos: 586.273212
interstitials-controller.ts:1135 [log] > [interstitials]: setSchedulePosition 1, undefined ([primary: 210.69-735.66]) pos: 586.273212
interstitials-controller.ts:2005 [log] > [interstitials]: INTERSTITIALS_BUFFERED_TO_BOUNDARY [primary: 210.69-735.66] (Infinity remaining)
hls.ts:556 [log] > startLoad(210.68799999009497)
base-playlist-controller.ts:346 [log] > [level-controller]: reload live playlist 0bps in 12651 ms
base-stream-controller.ts:2230 [log] > [stream-controller]: IDLE->STOPPED
base-stream-controller.ts:2230 [log] > [stream-controller]: STOPPED->IDLE
base-stream-controller.ts:965 [log] > [stream-controller]: Loading main sn: 11537 of level 0 (frag:[583.595-593.408]) cc: 0 [11477-13328], target: 586.273
base-stream-controller.ts:2230 [log] > [stream-controller]: IDLE->FRAG_LOADING
bcc49293-4b96-4d2b-84ff-ad12f89764bd:1168 [log] > Debug logs enabled for "main" in hls.js version 1.6.12
base-stream-controller.ts:2230 [log] > [subtitle-stream-controller]: IDLE->STOPPED
base-stream-controller.ts:2230 [log] > [subtitle-stream-controller]: STOPPED->IDLE
interstitials-controller.ts:1397 [log] > [interstitials]: resuming [primary: 210.69-735.66]
interstitials-controller.ts:1487 [log] > [interstitials]: [attachPrimary] Advancing timeline position to 586.273212
interstitials-controller.ts:1420 [log] > [interstitials]: INTERSTITIALS_PRIMARY_RESUMED [primary: 210.69-735.66]
base-stream-controller.ts:355 [log] > [stream-controller]: Media seeking to 586.273, state: FRAG_LOADING, out of buffer
base-stream-controller.ts:355 [log] > [audio-stream-controller]: Media seeking to 586.273, state: STOPPED, out of buffer
base-stream-controller.ts:355 [log] > [subtitle-stream-controller]: Media seeking to 586.273, state: IDLE, out of buffer
transmuxer-interface.ts:238 [log] > [transmuxer-interface]: Starting new transmux session for main sn: 11537 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 583.5946666570948
        initSegmentChange: true
base-stream-controller.ts:537 [log] > [stream-controller]: Loaded main sn: 11537 of level 0
transmuxer-interface.ts:396 [log] > ADTS sync word found !
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: reset next timestamp
bcc49293-4b96-4d2b-84ff-ad12f89764bd:11091 [log] > manifest codec:undefined, parsed codec:mp4a.40.2, channels:2, rate:48000 (ADTS object type:2 sampling index:3)
base-stream-controller.ts:2230 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1490 [log] > [stream-controller]: Init audio buffer, container:audio/mp4, codecs[selected/level/parsed]=[mp4a.40.2//mp4a.40.2]
buffer-controller.ts:594 [log] > [buffer-controller]: BUFFER_CODECS: "audio" (current SB count 1)
buffer-controller.ts:952 [log] > [buffer-controller]: queuing "audio" append sn: 11537 of level 0 cc: 0
transmuxer-interface.ts:396 [log] > [transmuxer.ts]: Flushed main sn: 11537 of level 0
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: ISGenerated flag reset
stream-controller.ts:1490 [log] > [stream-controller]: Init audio buffer, container:audio/mp4, codecs[selected/level/parsed]=[mp4a.40.2//mp4a.40.2]
buffer-controller.ts:594 [log] > [buffer-controller]: BUFFER_CODECS: "audio" (current SB count 1)
buffer-controller.ts:952 [log] > [buffer-controller]: queuing "audio" append sn: 11537 of level 0 cc: 0
base-stream-controller.ts:2160 [warn] > [stream-controller]: Found no media in fragment 11537 of level 0 resetting transmuxer to fallback to playlist timing
updateLevelTiming @ base-stream-controller.ts:2160
_handleTransmuxerFlush @ base-stream-controller.ts:1109
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389
base-stream-controller.ts:2230 [log] > [stream-controller]: PARSING->ERROR
main.js:745 Error event: {type: 'mediaError', details: 'fragParsingError', fatal: false, error: Error: Found no media in fragment 11537 of level 0 resetting transmuxer to fallback to playlist tim…, frag: Fragment, …}
(anonymous) @ main.js:745
emit @ index.js:203
emit @ hls.ts:386
trigger @ hls.ts:394
updateLevelTiming @ base-stream-controller.ts:2161
_handleTransmuxerFlush @ base-stream-controller.ts:1109
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389
base-stream-controller.ts:2230 [log] > [stream-controller]: ERROR->PARSED
base-stream-controller.ts:2177 [log] > [stream-controller]: Parsed main sn: 11537 of level 0 (frag:[583.595-593.408])
buffer-controller.ts:1034 [warn] > [buffer-controller]: Fragments must have at least one ElementaryStreamType set. type: main level: 0 sn: 11537
onFragParsed @ buffer-controller.ts:1034
emit @ index.js:203
emit @ hls.ts:386
trigger @ hls.ts:394
updateLevelTiming @ base-stream-controller.ts:2182
_handleTransmuxerFlush @ base-stream-controller.ts:1109
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389
buffer-controller.ts:1738 [log] > [buffer-controller]: Blocking operation requested, but no SourceBuffers exist
base-stream-controller.ts:768 [log] > [stream-controller]: Buffered main sn: 11537 of level 0 (frag:[583.595-593.408] > buffer:[17968.213-18007.467])
base-stream-controller.ts:2230 [log] > [stream-controller]: PARSED->IDLE
bcc49293-4b96-4d2b-84ff-ad12f89764bd:1168 [log] > Debug logs enabled for "main" in hls.js version 1.6.12
base-stream-controller.ts:965 [log] > [stream-controller]: Loading main sn: 11538 of level 0 (frag:[593.408-603.221]) cc: 0 [11477-13328], target: 593.408
base-stream-controller.ts:2230 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:238 [log] > [transmuxer-interface]: Starting new transmux session for main sn: 11538 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 593.4079999900947
        initSegmentChange: true
base-stream-controller.ts:537 [log] > [stream-controller]: Loaded main sn: 11538 of level 0
transmuxer-interface.ts:396 [log] > ADTS sync word found !
bcc49293-4b96-4d2b-84ff-ad12f89764bd:11091 [log] > manifest codec:undefined, parsed codec:mp4a.40.2, channels:2, rate:48000 (ADTS object type:2 sampling index:3)
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: reset next timestamp
base-stream-controller.ts:2230 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1490 [log] > [stream-controller]: Init audio buffer, container:audio/mp4, codecs[selected/level/parsed]=[mp4a.40.2//mp4a.40.2]
buffer-controller.ts:594 [log] > [buffer-controller]: BUFFER_CODECS: "audio" (current SB count 1)
buffer-controller.ts:952 [log] > [buffer-controller]: queuing "audio" append sn: 11538 of level 0 cc: 0
transmuxer-interface.ts:396 [log] > [transmuxer.ts]: Flushed main sn: 11538 of level 0
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: ISGenerated flag reset
stream-controller.ts:1490 [log] > [stream-controller]: Init audio buffer, container:audio/mp4, codecs[selected/level/parsed]=[mp4a.40.2//mp4a.40.2]
buffer-controller.ts:594 [log] > [buffer-controller]: BUFFER_CODECS: "audio" (current SB count 1)
buffer-controller.ts:952 [log] > [buffer-controller]: queuing "audio" append sn: 11538 of level 0 cc: 0
base-stream-controller.ts:2160 [warn] > [stream-controller]: Found no media in fragment 11538 of level 0 resetting transmuxer to fallback to playlist timing
updateLevelTiming @ base-stream-controller.ts:2160
_handleTransmuxerFlush @ base-stream-controller.ts:1109
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389
content-steering-controller.ts:225 [warn] > [content-steering]: Could not resolve fragParsingError ("Found no media in fragment 11538 of level 0 resetting transmuxer to fallback to playlist timing") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":7726.100000143051}
onError @ content-steering-controller.ts:225
emit @ index.js:203
emit @ hls.ts:386
trigger @ hls.ts:394
updateLevelTiming @ base-stream-controller.ts:2161
_handleTransmuxerFlush @ base-stream-controller.ts:1109
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389
base-stream-controller.ts:2230 [log] > [stream-controller]: PARSING->IDLE
main.js:745 Error event: {type: 'mediaError', details: 'fragParsingError', fatal: false, error: Error: Found no media in fragment 11538 of level 0 resetting transmuxer to fallback to playlist tim…, frag: Fragment, …}details: "fragParsingError"error: Error: Found no media in fragment 11538 of level 0 resetting transmuxer to fallback to playlist timing
    at StreamController.updateLevelTiming (https://hlsjs.video-dev.org/dist/hls.js:10742:23)
    at StreamController._handleTransmuxerFlush (https://hlsjs.video-dev.org/dist/hls.js:10017:12)
    at TransmuxerInterface.onWorkerMessage (https://hlsjs.video-dev.org/dist/hls.js:17081:21)errorAction: {action: 2, flags: 1, retryConfig: {…}, retryCount: 1, resolved: true}fatal: falsefrag: Fragment {_byteRange: null, _url: '', _stats: LoadStats, _streams: {…}, base: {…}, …}reason: "Found no media in msn 11538 of level \"?CMCD=\""type: "mediaError"[[Prototype]]: Object
(anonymous) @ main.js:745
emit @ index.js:203
emit @ hls.ts:386
trigger @ hls.ts:394
updateLevelTiming @ base-stream-controller.ts:2161
_handleTransmuxerFlush @ base-stream-controller.ts:1109
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389
base-stream-controller.ts:2230 [log] > [stream-controller]: IDLE->PARSED
base-stream-controller.ts:2177 [log] > [stream-controller]: Parsed main sn: 11538 of level 0 (frag:[593.408-603.221])
buffer-controller.ts:1034 [warn] > [buffer-controller]: Fragments must have at least one ElementaryStreamType set. type: main level: 0 sn: 11538
onFragParsed @ buffer-controller.ts:1034
emit @ index.js:203
emit @ hls.ts:386
trigger @ hls.ts:394
updateLevelTiming @ base-stream-controller.ts:2182
_handleTransmuxerFlush @ base-stream-controller.ts:1109
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389
buffer-controller.ts:1738 [log] > [buffer-controller]: Blocking operation requested, but no SourceBuffers exist
base-stream-controller.ts:768 [log] > [stream-controller]: Buffered main sn: 11538 of level 0 (frag:[593.408-603.221] > buffer:[17968.213-18007.467])
base-stream-controller.ts:2230 [log] > [stream-controller]: PARSED->IDLE
base-stream-controller.ts:1439 [log] > [stream-controller]: buffer full after gaps in "main" playlist starting at sn: 11539
bcc49293-4b96-4d2b-84ff-ad12f89764bd:1168 [log] > Debug logs enabled for "main" in hls.js version 1.6.12
level-controller.ts:655 [log] > [level-controller]: Loading level index 0 age 19.7 
base-playlist-controller.ts:216 [log] > [level-controller]: live playlist 0 REFRESHED 13330--1
base-playlist-controller.ts:346 [log] > [level-controller]: reload live playlist 0bps in 9922 ms
stream-controller.ts:661 [log] > [stream-controller]: Level 0 loaded [11479,13330][part-13330--1], cc [0, 0] duration:18007.44533334324
interstitials-controller.ts:1693 [log] > [interstitials]: INTERSTITIALS_UPDATED (19): ["1757586580_1095971179" 19.63-230.31],["1757587483_828786525" 735.66-915.63],["1757588208_3275627352" 1461.21-1641.13],["1757589360_1112304279" 2612.95-2735.27],["1757590199_3323689064" 3451.48-3661.48],["1757591128_2352494204" 4380.59-4560.55],["1757591808_3217355483" 5061.03-5241.00],["1757592868_650490437" 6120.70-6241.02],["1757593900_2452591813" 7152.51-7362.22],["1757594765_219942117" 8017.62-8197.59],["1757595408_4186765737" 8660.91-8840.83],["1757596540_3609787140" 9792.77-9912.70],["1757597300_3243867178" 10553.13-10763.09],["1757598283_2210540883" 11535.83-11715.67],["1757599008_1397443122" 12261.16-12441.17],["1757600262_4026601524" 13515.14-13727.32],["1757601780_209283036" 15033.00-15334.74],["1757602548_735391794" 15801.00-16040.96],["1757603906_4026256504" 17158.87-17369.60]
Schedule: [primary: 0.00-19.63],["1757586580_1095971179": 19.63-230.31],[primary: 230.31-735.66],["1757587483_828786525": 735.66-915.63],[primary: 915.63-1461.21],["1757588208_3275627352": 1461.21-1641.13],[primary: 1641.13-2612.95],["1757589360_1112304279": 2612.95-2735.27],[primary: 2735.27-3451.48],["1757590199_3323689064": 3451.48-3661.48],[primary: 3661.48-4380.59],["1757591128_2352494204": 4380.59-4560.55],[primary: 4560.55-5061.03],["1757591808_3217355483": 5061.03-5241.00],[primary: 5241.00-6120.70],["1757592868_650490437": 6120.70-6241.02],[primary: 6241.02-7152.51],["1757593900_2452591813": 7152.51-7362.22],[primary: 7362.22-8017.62],["1757594765_219942117": 8017.62-8197.59],[primary: 8197.59-8660.91],["1757595408_4186765737": 8660.91-8840.83],[primary: 8840.83-9792.77],["1757596540_3609787140": 9792.77-9912.70],[primary: 9912.70-10553.13],["1757597300_3243867178": 10553.13-10763.09],[primary: 10763.09-11535.83],["1757598283_2210540883": 11535.83-11715.67],[primary: 11715.67-12261.16],["1757599008_1397443122": 12261.16-12441.17],[primary: 12441.17-13515.14],["1757600262_4026601524": 13515.14-13727.32],[primary: 13727.32-15033.00],["1757601780_209283036": 15033.00-15334.74],[primary: 15334.74-15801.00],["1757602548_735391794": 15801.00-16040.96],[primary: 16040.96-17158.87],["1757603906_4026256504": 17158.87-17369.60],[primary: 17369.60-Infinity] pos: 586.273212
buffer-controller.ts:1352 [log] > [buffer-controller]: Updating MediaSource duration to 18027.072
base-stream-controller.ts:965 [log] > [stream-controller]: Loading main sn: 11537 of level 0 (frag:[583.595-593.408]) cc: 0 [11479-13330], target: 586.273
base-stream-controller.ts:2230 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:238 [log] > [transmuxer-interface]: Starting new transmux session for main sn: 11537 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 583.5946666570948
        initSegmentChange: true
base-stream-controller.ts:537 [log] > [stream-controller]: Loaded main sn: 11537 of level 0
bcc49293-4b96-4d2b-84ff-ad12f89764bd:11091 [log] > manifest codec:undefined, parsed codec:mp4a.40.2, channels:2, rate:48000 (ADTS object type:2 sampling index:3)
transmuxer-interface.ts:396 [log] > ADTS sync word found !
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: reset next timestamp
base-stream-controller.ts:2230 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1490 [log] > [stream-controller]: Init audio buffer, container:audio/mp4, codecs[selected/level/parsed]=[mp4a.40.2//mp4a.40.2]
buffer-controller.ts:594 [log] > [buffer-controller]: BUFFER_CODECS: "audio" (current SB count 1)
buffer-controller.ts:952 [log] > [buffer-controller]: queuing "audio" append sn: 11537 of level 0 cc: 0
transmuxer-interface.ts:396 [log] > [transmuxer.ts]: Flushed main sn: 11537 of level 0
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: ISGenerated flag reset
stream-controller.ts:1490 [log] > [stream-controller]: Init audio buffer, container:audio/mp4, codecs[selected/level/parsed]=[mp4a.40.2//mp4a.40.2]
buffer-controller.ts:594 [log] > [buffer-controller]: BUFFER_CODECS: "audio" (current SB count 1)
buffer-controller.ts:952 [log] > [buffer-controller]: queuing "audio" append sn: 11537 of level 0 cc: 0
base-stream-controller.ts:2160 [warn] > [stream-controller]: Found no media in fragment 11537 of level 0 resetting transmuxer to fallback to playlist timing
updateLevelTiming @ base-stream-controller.ts:2160
_handleTransmuxerFlush @ base-stream-controller.ts:1109
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389
content-steering-controller.ts:225 [warn] > [content-steering]: Could not resolve fragParsingError ("Found no media in fragment 11537 of level 0 resetting transmuxer to fallback to playlist timing") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":7726.100000143051}
onError @ content-steering-controller.ts:225
emit @ index.js:203
emit @ hls.ts:386
trigger @ hls.ts:394
updateLevelTiming @ base-stream-controller.ts:2161
_handleTransmuxerFlush @ base-stream-controller.ts:1109
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389
base-stream-controller.ts:2230 [log] > [stream-controller]: PARSING->IDLE
main.js:745 Error event: {type: 'mediaError', details: 'fragParsingError', fatal: false, error: Error: Found no media in fragment 11537 of level 0 resetting transmuxer to fallback to playlist tim…, frag: Fragment, …}
(anonymous) @ main.js:745
emit @ index.js:203
emit @ hls.ts:386
trigger @ hls.ts:394
updateLevelTiming @ base-stream-controller.ts:2161
_handleTransmuxerFlush @ base-stream-controller.ts:1109
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389
base-stream-controller.ts:2230 [log] > [stream-controller]: IDLE->PARSED
base-stream-controller.ts:2177 [log] > [stream-controller]: Parsed main sn: 11537 of level 0 (frag:[583.595-593.408])
buffer-controller.ts:1034 [warn] > [buffer-controller]: Fragments must have at least one ElementaryStreamType set. type: main level: 0 sn: 11537
onFragParsed @ buffer-controller.ts:1034
emit @ index.js:203
emit @ hls.ts:386
trigger @ hls.ts:394
updateLevelTiming @ base-stream-controller.ts:2182
_handleTransmuxerFlush @ base-stream-controller.ts:1109
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389
buffer-controller.ts:1738 [log] > [buffer-controller]: Blocking operation requested, but no SourceBuffers exist
bcc49293-4b96-4d2b-84ff-ad12f89764bd:1168 [log] > Debug logs enabled for "main" in hls.js version 1.6.12
base-stream-controller.ts:768 [log] > [stream-controller]: Buffered main sn: 11537 of level 0 (frag:[583.595-593.408] > buffer:[17968.213-18007.467])
base-stream-controller.ts:2230 [log] > [stream-controller]: PARSED->IDLE
base-stream-controller.ts:965 [log] > [stream-controller]: Loading main sn: 11538 of level 0 (frag:[593.408-603.221]) cc: 0 [11479-13330], target: 593.408
base-stream-controller.ts:2230 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:238 [log] > [transmuxer-interface]: Starting new transmux session for main sn: 11538 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 593.4079999900947
        initSegmentChange: true
base-stream-controller.ts:537 [log] > [stream-controller]: Loaded main sn: 11538 of level 0
transmuxer-interface.ts:396 [log] > ADTS sync word found !
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: reset next timestamp
bcc49293-4b96-4d2b-84ff-ad12f89764bd:11091 [log] > manifest codec:undefined, parsed codec:mp4a.40.2, channels:2, rate:48000 (ADTS object type:2 sampling index:3)
base-stream-controller.ts:2230 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1490 [log] > [stream-controller]: Init audio buffer, container:audio/mp4, codecs[selected/level/parsed]=[mp4a.40.2//mp4a.40.2]
buffer-controller.ts:594 [log] > [buffer-controller]: BUFFER_CODECS: "audio" (current SB count 1)
buffer-controller.ts:952 [log] > [buffer-controller]: queuing "audio" append sn: 11538 of level 0 cc: 0
transmuxer-interface.ts:396 [log] > [transmuxer.ts]: Flushed main sn: 11538 of level 0
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: ISGenerated flag reset
stream-controller.ts:1490 [log] > [stream-controller]: Init audio buffer, container:audio/mp4, codecs[selected/level/parsed]=[mp4a.40.2//mp4a.40.2]
buffer-controller.ts:594 [log] > [buffer-controller]: BUFFER_CODECS: "audio" (current SB count 1)
buffer-controller.ts:952 [log] > [buffer-controller]: queuing "audio" append sn: 11538 of level 0 cc: 0
base-stream-controller.ts:2160 [warn] > [stream-controller]: Found no media in fragment 11538 of level 0 resetting transmuxer to fallback to playlist timing
updateLevelTiming @ base-stream-controller.ts:2160
_handleTransmuxerFlush @ base-stream-controller.ts:1109
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389
content-steering-controller.ts:225 [warn] > [content-steering]: Could not resolve fragParsingError ("Found no media in fragment 11538 of level 0 resetting transmuxer to fallback to playlist timing") with content-steering for Pathway: . levels: 1 priorities: ["."] penalized: {".":7726.100000143051}
onError @ content-steering-controller.ts:225
emit @ index.js:203
emit @ hls.ts:386
trigger @ hls.ts:394
updateLevelTiming @ base-stream-controller.ts:2161
_handleTransmuxerFlush @ base-stream-controller.ts:1109
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389
base-stream-controller.ts:2230 [log] > [stream-controller]: PARSING->IDLE
main.js:745 Error event: {type: 'mediaError', details: 'fragParsingError', fatal: false, error: Error: Found no media in fragment 11538 of level 0 resetting transmuxer to fallback to playlist tim…, frag: Fragment, …}
(anonymous) @ main.js:745
emit @ index.js:203
emit @ hls.ts:386
trigger @ hls.ts:394
updateLevelTiming @ base-stream-controller.ts:2161
_handleTransmuxerFlush @ base-stream-controller.ts:1109
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389
base-stream-controller.ts:2230 [log] > [stream-controller]: IDLE->PARSED
base-stream-controller.ts:2177 [log] > [stream-controller]: Parsed main sn: 11538 of level 0 (frag:[593.408-603.221])
buffer-controller.ts:1034 [warn] > [buffer-controller]: Fragments must have at least one ElementaryStreamType set. type: main level: 0 sn: 11538
onFragParsed @ buffer-controller.ts:1034
emit @ index.js:203
emit @ hls.ts:386
trigger @ hls.ts:394
updateLevelTiming @ base-stream-controller.ts:2182
_handleTransmuxerFlush @ base-stream-controller.ts:1109
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389
```

### Chrome media internals output

```shell

```

--- a/src/remux/mp4-remuxer.ts
+++ b/src/remux/mp4-remuxer.ts
@@ -4,7 +4,10 @@ import { ErrorDetails, ErrorTypes } from '../errors';
 import { Events } from '../events';
 import { PlaylistLevelType } from '../types/loader';
 import { type ILogger, Logger } from '../utils/logger';
-import { toMsFromMpegTsClock } from '../utils/timescale-conversion';
+import {
+  timestampToString,
+  toMsFromMpegTsClock,
+} from '../utils/timescale-conversion';
 import type { HlsConfig } from '../config';
 import type { HlsEventEmitter } from '../events';
 import type { SourceBufferName } from '../types/buffer';
@@ -107,7 +110,19 @@ export default class MP4Remuxer extends Logger implements Remuxer {
   }
 
   resetTimeStamp(defaultTimeStamp: TimestampOffset | null) {
-    this.log('initPTS & initDTS reset');
+    const initPTS = this._initPTS;
+    if (
+      !initPTS ||
+      !defaultTimeStamp ||
+      defaultTimeStamp.trackId !== initPTS.trackId ||
+      defaultTimeStamp.baseTime !== initPTS.baseTime ||
+      defaultTimeStamp.timescale !== initPTS.timescale
+    ) {
+      this.log(
+        `Reset initPTS: ${initPTS ? timestampToString(initPTS) : initPTS} > ${defaultTimeStamp ? timestampToString(defaultTimeStamp) : defaultTimeStamp}`,
+      );
+    }
+
     this._initPTS = this._initDTS = defaultTimeStamp;
   }
 
@@ -334,6 +349,25 @@ export default class MP4Remuxer extends Logger implements Remuxer {
     };
   }
 
+  computeInitPts(
+    basetime: number,
+    timescale: number,
+    presentationTime: number,
+    type: 'audio' | 'video',
+  ): number {
+    const offset = Math.round(presentationTime * timescale);
+    let timestamp = normalizePts(basetime, offset);
+    if (timestamp < offset + timescale) {
+      this.log(
+        `Adjusting PTS for rollover in timeline near ${(offset - timestamp) / timescale} ${type}`,
+      );
+      while (timestamp < offset + timescale) {
+        timestamp += 8589934592;
+      }
+    }
+    return timestamp - offset;
+  }
+
   generateIS(
     audioTrack: DemuxedAudioTrack,
     videoTrack: DemuxedVideoTrack,
@@ -395,8 +429,12 @@ export default class MP4Remuxer extends Logger implements Remuxer {
         timescale = audioTrack.inputTimeScale;
         if (!_initPTS || timescale !== _initPTS.timescale) {
           // remember first PTS of this demuxing context. for audio, PTS = DTS
-          initPTS = initDTS =
-            audioSamples[0].pts - Math.round(timescale * timeOffset);
+          initPTS = initDTS = this.computeInitPts(
+            audioSamples[0].pts,
+            timescale,
+            timeOffset,
+            'audio',
+          );
         } else {
           computePTSDTS = false;
         }
@@ -421,13 +459,22 @@ export default class MP4Remuxer extends Logger implements Remuxer {
         trackId = videoTrack.id;
         timescale = videoTrack.inputTimeScale;
         if (!_initPTS || timescale !== _initPTS.timescale) {
-          const startPTS = this.getVideoStartPts(videoSamples);
-          const startOffset = Math.round(timescale * timeOffset);
-          initDTS = Math.min(
-            initDTS as number,
-            normalizePts(videoSamples[0].dts, startPTS) - startOffset,
+          const basePTS = this.getVideoStartPts(videoSamples);
+          const baseDTS = normalizePts(videoSamples[0].dts, basePTS);
+          const videoInitDTS = this.computeInitPts(
+            baseDTS,
+            timescale,
+            timeOffset,
+            'video',
           );
-          initPTS = Math.min(initPTS as number, startPTS - startOffset);
+          const videoInitPTS = this.computeInitPts(
+            basePTS,
+            timescale,
+            timeOffset,
+            'video',
+          );
+          initDTS = Math.min(initDTS as number, videoInitDTS);
+          initPTS = Math.min(initPTS as number, videoInitPTS);
         } else {
           computePTSDTS = false;
         }
@@ -897,10 +944,17 @@ export default class MP4Remuxer extends Logger implements Remuxer {
     });
 
     if (!contiguous || nextAudioTs < 0) {
+      const sampleCount = inputSamples.length;
       // filter out sample with negative PTS that are not playable anyway
       // if we don't remove these negative samples, they will shift all audio samples forward.
       // leading to audio overlap between current / next fragment
       inputSamples = inputSamples.filter((sample) => sample.pts >= 0);
+
+      if (sampleCount !== inputSamples.length) {
+        this.warn(
+          `Removed ${inputSamples.length - sampleCount} of ${sampleCount} samples (initPTS ${initTime} / ${inputTimeScale})`,
+        );
+      }
 
       // in case all samples have negative PTS, and have been filtered out, return now
       if (!inputSamples.length) {

--- a/src/utils/timescale-conversion.ts
+++ b/src/utils/timescale-conversion.ts
@@ -39,3 +39,8 @@ export function toMpegTsClockFromTimescale(
 ): number {
   return toTimescaleFromBase(baseTime, MPEG_TS_CLOCK_FREQ_HZ, 1 / srcScale);
 }
+
+export function timestampToString(timestamp: TimestampOffset): string {
+  const { baseTime, timescale, trackId } = timestamp;
+  return `${baseTime / timescale} (${baseTime}/${timescale}) trackId: ${trackId}`;
+}

